internal/controllers/register.go

```diff
@@ -0,0 +1,76 @@
+package controllers
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+
+	"github.com/lib/pq"
+
+	"github.com/nextlag/gomart/internal/mw/auth"
+	"github.com/nextlag/gomart/pkg/generatestring"
+)
+
+// Register обрабатывает HTTP-запросы для регистрации пользователя.
+func (c Controller) Register(w http.ResponseWriter, r *http.Request) {
```

Comment:
```suggestion
func (c *Controller) Register(w http.ResponseWriter, r *http.Request) {
```
В принципе, можно оставить value receiver, но при большом количестве обращений при вызове каждой функции структура Controller будет копироваться и съедать память.